tests/run/i3812.scala

```diff
@@ -0,0 +1,8 @@
+object Test {
+  def main(args: Array[String]): Unit = {
+    val x = 42
+    val Y = "42"
+
+    x match { case Y.toInt          => () } // ok
```

Comment:
This is an error in scalac, I don't think we should support it either

Reply:
Good point, a check for stable prefix is missing for patterns here.

Reply:
Now I'm a little unsure about Scalac behavior, as it also rejects the following code:

```
    var X = 42
    42 match { case X          => () }
```

It seems to be it can be accepted without any issue.

Reply:
According to scala spec 2.13, the scalac behavior is correct.

https://www.scala-lang.org/files/archive/spec/2.13/08-pattern-matching.html#stable-identifier-patterns